src/dependency-checks.js

```diff
@@ -0,0 +1,265 @@
+const debug = require('debug')('ember-m3');
+const pkg = require('../package.json');
+const semver = require('semver');
+const M3DataDeps = Object.keys(pkg.dependencies).filter(
+  k => k.indexOf('ember-data') !== -1 || k === 'ember-inflector'
+);
+
+const REQUIRED_DATA_PACKAGES = ['@ember-data/model', '@ember-data/store', 'ember-inflector'];
+const ALL_DATA_PACKAGES = [
+  '@ember-data/adapter',
+  '@ember-data/debug',
+  '@ember-data/model',
+  '@ember-data/record-data',
+  '@ember-data/serializer',
+  '@ember-data/store',
+];
+const minVersionForUsingOwnPackages = '3.14.0';
+
+class VersionChecker {
+  constructor(project) {
+    this._project = project;
+  }
+
+  *allAddons() {
+    let cache = this._project.packageInfoCache.entries;
+    let keys = Object.keys(cache);
+    for (let key of keys) {
+      let packageInfo = cache[key];
+
+      if (
+        packageInfo.pkg &&
+        packageInfo.pkg.keywords &&
+        packageInfo.pkg.keywords.includes('ember-addon')
+      ) {
+        yield {
+          name: packageInfo.pkg.name,
+          pkg: packageInfo.pkg,
+          _info: packageInfo,
+        };
+      }
+    }
+  }
+
+  filterAddonsByName(name) {
+    const addons = [];
+
+    for (let addon of this.allAddons()) {
+      if (addon.name === name) {
+        addons.push(addon);
+      }
+    }
+
+    return addons;
+  }
+  static forProject(project) {
+    return new VersionChecker(project);
+  }
+}
+
+function extractVersionsFromDeps(deps, m3Version, dataInfo) {
```

Comment:
This is a pretty complex function, can you please add a comment explaining what it does. Maybe an example input -> output

Reply:
Added